### PR TITLE
Feature/basket

### PR DESCRIPTION
**Pull request by mstrutt**

### CODE REVIEW

File: app/assets/js/controllers/basket.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+function BasketController ($scope, basketService, universal_variable, debounceOptions) {
+	var vm = this;
+
+	vm.debounceOptions = debounceOptions;
+
+	vm.updateLineItem = function(line_item, index) {
+		universal_variable.basket.line_items[index].quantity = line_item.quantity;
```
Comment (toboid) [inline app/assets/js/controllers/basket.js:7]: Is there a reason to use snake case for the universal_variable service? Something to do with sending it up to the server?

File: app/assets/js/services/basket.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+			method: 'GET',
+			url: API.baskets + '/' + basket_id
+		})
+			.then(function(basket) {
+				universal_variable.basket = basket;
+				return basket;
+			});
+	};
+
+	service
```
Comment (toboid) [inline app/assets/js/services/basket.js]: `getBasket()` suggests that this function just gets a basket but I think it's being used for side effects here. Might be able to move the part that checks if the basket exists to `createBasket()` or extract to a separate method?

File: app/assets/js/controllers/basket.js (hunk 1 above)
Comment (mstrutt) [inline app/assets/js/controllers/basket.js:7]: just feels a bit odd writing it as `universalVariable` after being so use to seeing it called `universal_variable` in all the docs and such

File: app/assets/js/services/basket.js (hunk 2 above)
Comment (mstrutt) [inline app/assets/js/services/basket.js]: updated with an initiBasket method to avoid crossing the other two

**Pull request merged**